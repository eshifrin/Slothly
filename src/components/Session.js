import React, { Component } from 'react';
import { StyleSheet, View, Text, TouchableHighlight } from 'react-native';
import Separator from './Separator';
import { colors } from '../constants';
import RenderIf from '../utils/RenderIf';
import { formatDate } from '../utils/calcStats';
import { UpIcon, DownIcon } from '../utils/Icons';
import PropTypes from 'prop-types';

export default class Session extends Component {
  state = {
    showNote: false
  };

  render() {
    const { item } = this.props;
    return (
      <View style={styles.container}>
        <View style={styles.rowContainer}>
          <Text style={styles.rowText}>{item.sessionLength}</Text>
          <Text style={styles.rowText}>{formatDate(item.when)}</Text>
          <TouchableHighlight
            onPress={() => this.setState({ showNote: !this.state.showNote })}
            underlayColor="transparent"
          >
            <Text>{this.state.showNote ? UpIcon : DownIcon}</Text>
          </TouchableHighlight>
        </View>

        <RenderIf condition={this.state.showNote}>
          <View style={styles.noteContainer}>
            <Separator />
            <Text style={styles.noteHeader}>SESSION NOTES</Text>
            <Text style={styles.note}>{item.note}</Text>
          </View>
        </RenderIf>
      </View>
    );
  }
}

Session.propTypes = {
  item: PropTypes.shape({
    id: PropTypes.string,
    sessionLength: PropTypes.oneOfType(PropTypes.number, PropTypes.string),
    when: PropTypes.instanceOf(Date),
    note: PropTypes.string
  })
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'column',
    backgroundColor: colors.deepBlue
  },
  rowContainer: {
    flex: 1,
    justifyContent: 'space-around',
    flexDirection: 'row',
    alignItems: 'center',
    height: 75
  },
  rowText: {
    color: 'white',
    fontFamily: 'Futura',
    fontSize: 18
  },
  noteHeader: {
    fontWeight: '700',
    backgroundColor: colors.lightBLue,
    color: colors.deepBlue,
    fontFamily: 'Futura',
    fontSize: 13,
    flex: 1,
    height: 20,
    textAlign: 'center'
  },
  note: {
    backgroundColor: colors.lightBLue,
    color: colors.deepBlue,
    fontFamily: 'Futura',
    fontSize: 15,
    flex: 1
  },
  noteContainer: {
    backgroundColor: colors.lightBLue,

    flex: 1,
    flexDirection: 'column',
    paddingLeft: 15
  }
});
