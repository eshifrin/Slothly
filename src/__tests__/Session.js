import 'react-native';
import React from 'react';
import Session from '../components/Session';
import { shallow } from 'enzyme';

describe('Session', () => {
  let wrapper;
  const mockSession = {
    sessionLength: '5',
    when: new Date(2018, 0, 1),
    note: 'mock note'
  };

  beforeEach(done => {
    wrapper = shallow(<Session item={mockSession} />);
    setTimeout(() => done(), 0);
  });

  it('renders', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
