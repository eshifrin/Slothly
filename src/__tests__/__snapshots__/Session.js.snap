// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Session renders 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#032b82",
      "flex": 1,
      "flexDirection": "column",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "height": 75,
        "justifyContent": "space-around",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
          "fontFamily": "Futura",
          "fontSize": 18,
        }
      }
    >
      5
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
          "fontFamily": "Futura",
          "fontSize": 18,
        }
      }
    >
      Mon 12:00AM | Jan 1st, 2018
    </Text>
    <TouchableHighlight
      activeOpacity={0.85}
      onPress={[Function]}
      underlayColor="transparent"
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        <Icon
          allowFontScaling={false}
          color="white"
          name="chevron-down"
          size={25}
        />
      </Text>
    </TouchableHighlight>
  </View>
  <RenderIf
    condition={false}
  >
    <View
      style={
        Object {
          "backgroundColor": "#edf2f7",
          "flex": 1,
          "flexDirection": "column",
          "paddingLeft": 15,
        }
      }
    >
      <Separator />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "backgroundColor": "#edf2f7",
            "color": "#032b82",
            "flex": 1,
            "fontFamily": "Futura",
            "fontSize": 13,
            "fontWeight": "700",
            "height": 20,
            "textAlign": "center",
          }
        }
      >
        SESSION NOTES
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "backgroundColor": "#edf2f7",
            "color": "#032b82",
            "flex": 1,
            "fontFamily": "Futura",
            "fontSize": 15,
          }
        }
      >
        mock note
      </Text>
    </View>
  </RenderIf>
</View>
`;
